src/eventfd.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//TODO: Firecracker uses io::Error instead of sys_util::Error for Result<T, E>. Need discussion here.
+
+#![allow(unused)]
+use std::fs::File;
+use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
+use std::{io, mem, result};
+
+use libc::{c_void, dup, eventfd, read, write, EFD_NONBLOCK};
+
+/// A safe wrapper around a Linux eventfd (man 2 eventfd).
+///
+/// An eventfd is useful because it is sendable across processes and can be used for signaling in
+/// and out of the KVM API. They can also be polled like any other file descriptor.
+pub struct EventFd {
+    eventfd: File,
+}
+
+impl EventFd {
+    /// Creates a new blocking EventFd with an initial value of 0.
```

Comment:
I don't think this comment is accurate. We are creating the eventfd with EFD_NONBLOCK.
I noticed that in CrosVM they are opening it without the flag so maybe it would make sense to pass the additional flags as parameters.

Reply:
Yep. Let me change the flag to be a parameter for new() and test the new(EFD_NONBLOCK) in #[test]. Do you think we need double test the new(0) for each #[test]?

Reply:
Hmm, I'll look at the code. It is always nice to test all the usecases, now it depends whether it adds any value.